# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaStreamTrack.h
# Commit: 35d86243cb45
# Full Hash: 35d86243cb45f7e02dd6ec0eea04cd62c2a70196
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-16 09:14:26
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 7d143d92bf2a -r 35d86243cb45 dom/media/MediaStreamTrack.h
--- a/dom/media/MediaStreamTrack.h	Sun Sep 15 20:03:40 2024 +0000
+++ b/dom/media/MediaStreamTrack.h	Sun Sep 15 21:49:38 2024 +0000
@@ -40,6 +40,7 @@
 
 class AudioStreamTrack;
 class VideoStreamTrack;
+class RTCStatsTimestampMaker;
 enum class CallerType : uint32_t;
 
 /**
@@ -141,6 +142,14 @@
   virtual const PeerIdentity* GetPeerIdentity() const { return nullptr; }
 
   /**
+   * This is used in WebRTC. The timestampMaker can convert between different
+   * timestamp types used during the session.
+   */
+  virtual const RTCStatsTimestampMaker* GetTimestampMaker() const {
+    return nullptr;
+  }
+
+  /**
    * MediaStreamTrack::GetLabel (see spec) calls through to here.
    */
   void GetLabel(nsAString& aLabel) { aLabel.Assign(mLabel); }
@@ -459,6 +468,13 @@
     return GetSource().GetPeerIdentity();
   }
 
+  /**
+   * Get this track's RTCStatsTimestampMaker.
+   */
+  const RTCStatsTimestampMaker* GetTimestampMaker() const {
+    return GetSource().GetTimestampMaker();
+  }
+
   ProcessedMediaTrack* GetTrack() const;
   MediaTrackGraph* Graph() const;
   MediaTrackGraphImpl* GraphImpl() const;